rasa/nlu/extractors/mitie_entity_extractor.py

```diff
@@ -1,89 +1,130 @@
+        if num_threads is None or num_threads <= 0:
+            raise InvalidConfigException(
+                f"Expected `num_threads` to be some value >= 1 (default: 1)."
+                f"but received {num_threads}"
+            )
 
     @classmethod
-    def required_packages(cls) -> List[Text]:
-        return ["mitie"]
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> GraphComponent:
+        """Creates a new `MitieEntityExtractorGraphComponent`.
 
-    def extract_entities(
-        self,
-        text: Text,
-        tokens: List[Token],
-        feature_extractor: Optional["mitie.total_word_feature_extractor"],
-    ) -> List[Dict[Text, Any]]:
-        ents = []
-        tokens_strs = [token.text for token in tokens]
-        if self.ner:
-            entities = self.ner.extract_entities(tokens_strs, feature_extractor)
-            for e in entities:
-                if len(e[0]):
-                    start = tokens[e[0][0]].start
-                    end = tokens[e[0][-1]].end
-
-                    ents.append(
-                        {
-                            "entity": e[1],
-                            "value": text[start:end],
-                            "start": start,
-                            "end": end,
-                            "confidence": None,
-                        }
-                    )
+        Args:
+            config: This config overrides the `default_config`.
+            model_storage: Storage which graph components can use to persist and load
+                themselves.
+            resource: Resource locator for this component which can be used to persist
+                and load itself from the `model_storage`.
+            execution_context: Information about the current graph run. Unused.
 
-        return ents
+        Returns: An instantiated `MitieEntityExtractorGraphComponent`.
+        """
+        return cls(config, model_storage, resource)
 
-    def train(
-        self,
-        training_data: TrainingData,
-        config: Optional[RasaNLUModelConfig] = None,
-        **kwargs: Any,
-    ) -> None:
+    def train(self, training_data: TrainingData, mitie_model: MitieModel) -> Resource:
+        """Trains a MITIE named entity recognizer.
+
+        Args:
+            training_data: the training data
+            mitie_model: a MitieModel
+        Returns:
+            resource for loading the trained model
+        """
         import mitie
 
-        model_file = kwargs.get("mitie_file")
-        if not model_file:
-            raise Exception(
-                "Can not run MITIE entity extractor without a "
-                "language model. Make sure this component is "
-                "preceeded by the 'MitieNLP' component."
-            )
+        trainer = mitie.ner_trainer(str(mitie_model.model_path))
```

Comment:
aha so we can't actually re-use the loaded model? At this point the path could just be in the config I guess...

Reply:
we use it in process